week_09/case_study_09.Rmd

```diff
@@ -1,7 +1,57 @@
 ---
 title: "Case Study 09"
-author: Your Name
-date: August 1, 2020
+author: Hang Tian
+date: "`r Sys.Date()`"
 output: github_document
 ---
- 
\ No newline at end of file
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+set.seed(1) # seed for any random generation
+```
+# 1. Import packages and get data
+```{r}
+library(sf)
+library(tidyverse)
+library(ggmap)
+library(rnoaa)
+library(spData)
+data(world)
+data(us_states)
+# Download zipped data from noaa with storm track information
+dataurl="https://www.ncei.noaa.gov/data/international-best-track-archive-for-climate-stewardship-ibtracs/v04r00/access/shapefile/IBTrACS.NA.list.v04r00.points.zip"
+
+tdir=tempdir()
+download.file(dataurl,destfile=file.path(tdir,"temp.zip"))
+unzip(file.path(tdir,"temp.zip"),exdir = tdir) #unzip the compressed folder
+storm_data <- read_sf(list.files(tdir,pattern=".shp",full.names = T))
+```
+# 2. Data processing
+```{r}
+storm_cleaned=storm_data%>%filter(ISO_TIME>=1950)%>%
+  mutate_if(is.numeric, function(x) ifelse(x==-999.0,NA,x))%>%
+  mutate(decade=(floor(year/10)*10))
+region=st_bbox(storm_cleaned)
+
+```
+# 3. Plot
+```{r}
+ggplot(world)+
+  stat_bin2d(data=storm_cleaned,aes(y=st_coordinates(storm_cleaned)[,2], x=st_coordinates(storm_cleaned)[,1]),bins=100)+
```

Comment:
don't forget `geom_sf()` to add the country polygons before adding the hurricane tracks.